tfhe/src/integer/server_key/radix/comparison.rs

```diff
@@ -30,10 +31,10 @@ impl ServerKey {
     /// let ct_res = sks.unchecked_eq(&ct1, &ct2);
     ///
     /// // Decrypt:
-    /// let dec_result: u64 = cks.decrypt(&ct_res);
+    /// let dec_result: u64 = cks.decrypt_one_block(&ct_res.0);
```

Comment:
why can't decrypt be used here ?

Reply:
It should be `decrypt_bool`, to use decrypt I would need to convert the ct_res to a radix using .into_radix as ct_res is a BooleanBlock

Reply:
can't decrypt accept a BooleanBlock ? I would expect to get back 0 or 1 in that case

Reply:
I mean via some trait or mechanic to make it seamless

Reply:
I can do `pub trait IntegerUnsignedRadixCiphertext: IntegerRadixCiphertext {}` 
impl it on RadixCiphertext and BooleanValue, and use it as a bound to decrypt_radix.

It breaks usage of `cks.decrypt::<u64>(&value)`  they have to be changed to `cks.decrypt::<u64, _>(&value)`;

Is that ok ?


Sadly I can't do `pub trait IntegerUnsignedRadixCiphertext: IntegerRadixCiphertext<IS_SIGNED=false> {}` because  Is_SIGNED is a const generic not a type.


we could try to do 
```rust
pub trait SignednessMarker : sealed {
   const IS_SIGNED: bool;
}

pub struct UnsignedMarker;
pub struct SignedMarker;

impl SignednessMarker for SignedMarker {
  IS_SIGNED = true;
}

// same for unsigned

pub trait IntegerRadixCiphertext {
    type Signedness: SignednessMarker;
}


pub trait IntegerUnsignedRadixCiphertext: IntegerRadixCiphertext<Signedness=UnsignedMarker> {}
```

but its out of the scope of this pr so maybe later if it seems worth and doable

Reply:
IntegerDecryptableCiphertext maybe ?

Reply:
it's not critical right now but could be nice to handle that down the line

Reply:
`IntegerDecryptableCiphertext` is not explicit enough, decrypt_radix is meant to decrypt unsigned ciphertext

We have decrypt_radix for unsigned radix and decrypt_signed for signed radix, the seperation is there to avoid user decrypting a  mismatching clear

Reply:
We could have only one decrypt/encrypt function but that means users would have less safeguards , if they mismatch a type, it would be as if an implicit cast had been done

Reply:
ok yeah and implementing the IntegerRadixCipehrtext for Boolean defeats the whole purpose of the type

Reply:
I'm not sure yet, I'm just wondering if we have a way to make it a bit more uniform, I guess it may be possible, I just don't quite see it yet

Reply:
I already implemented IntegerRadixCipehrtext for BooleanValue.

Yes it means that it can be used in additions, making it non boolean anymore, (but with the way are signatures are, you can only add a boolean to a boolean)

But it also makes boolean usable in bitand/or/xor

The idea is that since we cannot verify if a ciphertext actually is boolean, it does not really matter if we allow addition/sub on booleans because in the end the user can wrap any ciphertext to be marked as boolean.


So to me the question is more: "does this bring enough value"

Reply:
If not, we could just go with Returning single shortint block / taking single shortint block when manipulating/returning boolean valued ciphertext and use documentation to tell when the returned value/input value has to be boolean

Reply:
I think the fact any ciphertext can be wrapped defeats the purpose unless we do a computation like value != 0 to store the boolean

Reply:
In that case I don't think introducing this is a good idea, because I could see cases where the users know the value in the ciphertext is either 0 or 1 and so making it pay the cost of a pbs for nothing does not sounds like a good idea

Reply:
so let's not do that change for now ? and revert the decrypt change ?

Reply:
:thinking:  so what should we keep ?

Reply:
you original one that did not change the decryption I think ?

Reply:
But that still has that any ciphertext can be wwrapper as boolean, and BooleanValue implements IntegerRadixCiphertext (if the wrapped thing is a radix)

Reply:
ah damn

Reply:
ok sorry maybe I did not follow enough what you wanted to do.

Why do we implement the trait for the Boolean thingy and why do we allow anything to be wrapped ?